homeassistant/components/zwave_js/discovery_data_template.py

```diff
@@ -507,64 +514,57 @@ def values_to_watch(self, resolved_data: dict[str, Any]) -> Iterable[ZwaveValue]
             resolved_data["configuration_value"],
         ]
 
-    def get_speed_config(
+    def get_fan_value_mapping(
         self, resolved_data: dict[str, ZwaveConfigurationValue]
-    ) -> list[int] | None:
-        """Get current speed configuration from resolved data."""
+    ) -> FanValueMapping | None:
+        """Get current fan properties from resolved data."""
         zwave_value: ZwaveValue = resolved_data["configuration_value"]
 
         if zwave_value.value is None:
-            _LOGGER.warning("Unable to read fan speed configuration value")
+            _LOGGER.warning("Unable to read fan configuration value")
             return None
 
-        speed_config = self.configuration_value_to_speeds.get(zwave_value.value)
-        if speed_config is None:
-            _LOGGER.warning("Unrecognized speed configuration value")
+        fan_value_mapping = self.configuration_value_to_fan_value_mapping.get(
+            zwave_value.value
+        )
+        if fan_value_mapping is None:
+            _LOGGER.warning("Unrecognized fan configuration value")
             return None
```

Comment:
same as above. Please add tests too to get coverage up to snuff. The best way to do this is to update the state dict of the fixture in the test and then create the node and attach it to the client manually (if I find an example of this I will add another comment)

Reply:
https://github.com/home-assistant/core/blob/ad4409bcb0e5f213ed9a2ddc1c5cdb93f6f50637/tests/components/zwave_js/test_sensor.py#L334:L386 is probably what you had in mind?

Reply:
yes exactly. You can modify the state to remove each value that causes a return `None` in two separate tests and validate that your fan entity works as expected

Reply:
Done!